### ISSUE TITLE
Allow guest users to set avatar

### ISSUE THREAD

**Issue by jancborchardt** [nextcloud/spreed#357]
Similar to how guests can set their name, it would be cool if they can upload an avatar. If we save that in the browser local storage (like the name), it can also be used for collaboration using Collabora, or other calls. :)

What do you think @nextcloud/spreed?

**Comment by nickvergessen** [comment]
And how would we store that on the server? there is no user id where we could put it in the avatars folder. Also guests get a new session for each visit, so we can not store it based on session. This means the guest/browser would need to upload it everytime again.
Since we then have to store it in a temporary spreed way (so we can delete them after the user timed out), there is at least no way for other apps to reuse this.

So not sure if it is worth the effort. I have never seen a webpage before, where guests can upload avatars without registering.

**Comment by jancborchardt** [comment]
How do we do it for the username, as this seems to be stored across calls and restarts?

I was thinking storing it in the browser local storage (as said above) might work, no?

**Comment by nickvergessen** [comment]
yeah that is send peer to peer on each call start and is stored in thr browser local storage I guess. That is also why the channel descriptions contain the guest count instead of the names (ref #216 ). php simply doesn't know about it and doesn't keep track of it.

**Comment by jancborchardt** [comment]
So in the same way we could save the image in the browser IndexedDB and send it peer-to-peer each call.

And since the local storage is domain-specific, both the guest username and avatar should also be possible to be used by the Collabora app. cc @timar
Even by say the activity/sharing app, if there's any tracking/display functionality of who downloaded a shared file. cc @schiessle

**Comment by nickvergessen** [comment]
To copy the comments from #1974 

* Guest avatars should also be colored
* Guest names must have a `(guest)` appended, similar to moderators, to prevent impersonation

**Comment by jancborchardt** [comment]
@nickvergessen this issue is specifically about setting images as avatars too though. :) Might be separate as that is more work?

**Comment by nickvergessen** [comment]
I dont rule out, those 2 things cant be done before, but they need to be made when this here happens :P
